colcon_defaults/argument_parser/defaults.py

```diff
@@ -24,6 +25,22 @@
     'line arguments (default: $COLCON_HOME/defaults.yaml)')
 
 
+def deep_update(source, overrides):
```

Comment:
Unless there is a compelling reason not to, let's prefix this with `_` to indicate that it isn't part of any public API.
```suggestion
def _deep_update(source, overrides):
```